### ISSUE TITLE
iOS 9 compatability/readiness?

### ISSUE THREAD

**Issue by ghost** [pokeb/asi-http-request#404]
Are there are any known issues with iOS9?

**Comment by jhollida24** [comment]
I'm curious about this as well. App Transport Security Layer requirements worry me.

**Comment by hirenj102** [comment]
Why asihttp,asiformdatarequest not working in iOS 9?Please help me

**Comment by wuzhupc** [comment]
ATS 
see:
https://developer.apple.com/library/prerelease/ios/technotes/App-Transport-Security-Technote/index.html#//apple_ref/doc/uid/TP40016240

**Comment by michalcichon** [comment]
Hi, for us, the library is generally working in iOS 9 (we have a project with deployment target set to 6.0) but have strange issues in iOS 9 with cutting responses. We have some services that gives us Base64 images in responses (responses are in JSON) but sometimes (randomly), JSONs are cut in random places. We checked whether the server responds correctly, but JSONs in revers proxy are OK.

Have anyone any idea what can be wrong? The library works fine in iOS 6, 7, 8 and in most cases (really, about 90% of our services) in iOS 9.

**Comment by pabloburnio** [comment]
@michalcichon - I'm also looking into intermitted issues with requests failing (seems to be since our release on ios9) - mainly for larger requests and for people on poorer connections. All traffic is over https in our case so wouldn't have thought it was ATS as I would have expected all, not just some requests to fail.

Did you manage to find anything at all? Appreciate any info - thanks.

**Comment by michalcichon** [comment]
@pabloburnio We checked whether it's the issue with decompression because our requests differed only in IDs (that we are randomising because of some security issues) but the decompression was OK. We found out that the length of the request is OK, but some requests had some last bits set to NULL. That's strange. 

It was not ATS issue but we cannot find connection between poor connection and the issue. We use https only and most of our services work great. The problem is only with one service with Base64 images.

**Comment by pabloburnio** [comment]
@michalcichon - Ah I see, thanks for the info anyway! Could be your base64 implementation, possibly not flushing the remaining bytes? Best of luck!

**Comment by michalcichon** [comment]
@pabloburnio Nope. Base64 is OK and it works on the different layer. ;) It's probably the issue with the library. We will probably change the lib to the other one, since we cannot find the root of the problem.

**Comment by chatvenkat** [comment]
hi i'm trying to use this in ios 9 and getting -[CFString release]: message sent to deallocated instance 0x7fe1b5b796f0

caused by this line when i segue to different view upon successful completion of this request.
[asiRequest setData:imageData1 withFileName:@"avatar.jpg" andContentType:@"image/jpeg" forKey:@"avatar"];

any one faced this issue ?
